pkg/cmd/kafka/create/data.go

```diff
@@ -0,0 +1,160 @@
+func GetValidKafkaSizes(f *factory.Factory,
+	providerID string, regionId string, amsType *accountmgmtutil.QuotaSpec) ([]string, error) {
+
+	conn, err := f.Connection(connection.DefaultConfigSkipMasAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	validSizes := []string{}
+
+	instanceTypes, _, err := conn.API().
+		KafkaMgmt().
+		GetInstanceTypesByCloudProviderAndRegion(f.Context, providerID, regionId).
+		Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	var desiredInstanceType string
+	if amsType != nil {
+		desiredInstanceType = mapAmsTypeToBackendType(amsType)
+	} else {
+		// If we do not know instance type from AMS we should assume standard instance type
+		// This is because trials do not have sizes so it is best to suggest all standard sizes instead
+		desiredInstanceType = mapAmsTypeToBackendType(
+			&accountmgmtutil.QuotaSpec{
+				Name:  accountmgmtutil.QuotaStandardType,
+				Quota: 1,
+			})
+	}
+
+	for _, instanceType := range instanceTypes.GetItems() {
+		if desiredInstanceType == instanceType.GetId() {
+			for _, instanceSize := range instanceType.GetSizes() {
+				validSizes = append(validSizes, instanceSize.GetId())
+			}
+		}
+	}
+	return validSizes, nil
+}
+
+// GetEnabledCloudProviderNames returns a list of cloud provider names from the enabled cloud providers
+func GetEnabledCloudProviderNames(f *factory.Factory) ([]string, error) {
+	validProviders := []string{}
+	conn, err := f.Connection(connection.DefaultConfigSkipMasAuth)
+	if err != nil {
+		return validProviders, err
+	}
+
+	cloudProviderResponse, _, err := conn.API().KafkaMgmt().GetCloudProviders(f.Context).Execute()
+	if err != nil {
+		return validProviders, err
+	}
+
+	cloudProviders := cloudProviderResponse.GetItems()
+	cloudProviderNames := []string{}
+	for _, provider := range cloudProviders {
+		if provider.GetEnabled() {
+			cloudProviderNames = append(cloudProviderNames, provider.GetName())
+		}
+	}
+	return cloudProviderNames, err
+}
+
+// FindCloudProviderByName finds and returns a cloud provider item from the list by its name
+func FindCloudProviderByName(cloudProviders []kafkamgmtclient.CloudProvider, name string) *kafkamgmtclient.CloudProvider {
+	for _, p := range cloudProviders {
+		if p.GetName() == name {
+			return &p
+		}
+	}
+	return nil
+}
+
+// GetEnabledCloudRegionIDs extracts and returns a slice of the unique IDs of all enabled regions
+func GetEnabledCloudRegionIDs(f *factory.Factory, providerID string, userAllowedAMSInstanceType *accountmgmtutil.QuotaSpec) ([]string, error) {
+	validRegions := []string{}
+	conn, err := f.Connection(connection.DefaultConfigSkipMasAuth)
+	if err != nil {
+		return validRegions, err
+	}
+
+	cloudProviderResponse, _, err := conn.API().
+		KafkaMgmt().
+		GetCloudProviderRegions(f.Context, providerID).
+		Execute()
+	if err != nil {
+		return validRegions, err
+	}
+
+	var serverInstanceType string
+	if userAllowedAMSInstanceType != nil {
```

Comment:
Would the backend implementing a filter by Account Management Service instance type help here?